datafusion/sql/src/expr/mod.rs

```diff
@@ -224,14 +225,23 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
             SQLExpr::Cast {
                 expr, data_type, ..
-            } => Ok(Expr::Cast(Cast::new(
-                Box::new(self.sql_expr_to_logical_expr(
-                    *expr,
-                    schema,
-                    planner_context,
-                )?),
-                self.convert_data_type(&data_type)?,
-            ))),
+            } => {
+                let mut dt = self.convert_data_type(&data_type)?;
+                let expr =
+                    self.sql_expr_to_logical_expr(*expr, schema, planner_context)?;
+
+                // int/floats should come as seconds rather as nanoseconds
+                dt = match &dt {
```

Comment:
It seems strange to special case `CAST AS TIMESTAMP` to produce a `Timestamp::Seconds` while anywhere else in SQL `TIMESTAMP` means `Timestamp::Nanoseconds`

Did you consider changing

https://github.com/apache/arrow-datafusion/blob/8fcc5e03147db93d688c2dfa112653191476f2bf/datafusion/sql/src/planner.rs#L335-L348

🤔